lib/configurators/aws/aws.go

```diff
@@ -316,6 +319,45 @@ type ssmClient interface {
 	CreateDocument(ctx context.Context, params *ssm.CreateDocumentInput, optFns ...func(*ssm.Options)) (*ssm.CreateDocumentOutput, error)
 }
 
+type localRegionGetter interface {
+	GetRegion(context.Context) (string, error)
+}
+
+func getFallbackRegion(ctx context.Context, w io.Writer, fips bool, localRegionGetter localRegionGetter) (string, error) {
+	if localRegionGetter == nil {
+		imdsClient, err := awsimds.NewInstanceMetadataClient(ctx)
+		if err == nil && imdsClient != nil && imdsClient.IsAvailable(ctx) {
+			localRegionGetter = imdsClient
+		}
+	}
+
+	if localRegionGetter != nil {
+		localRegion, err := localRegionGetter.GetRegion(ctx)
+		if err == nil && localRegion != "" {
+			fmt.Fprintf(w, "Using region %q from instance metadata.\n", localRegion)
+			return localRegion, nil
+		}
+	}
+
+	// Fips STS endpoints are regional.
+	if fips {
+		return "", trace.Errorf("no region found from the default AWS config or instance metadata. Please provide a region in your AWS config or through AWS_REGION environment variable.")
+	}
+
+	// Fallback to aws-global which works for both STS and IAM services. Print
+	// a warning since global STS endpoint is legacy and it won't work in
+	// non-standard partitions like us-gov-, cn- regions.
+	//
+	// Reference:
+	// https://docs.aws.amazon.com/sdkref/latest/guide/feature-sts-regionalized-endpoints.html
+	fmt.Fprint(w, `
+Warning. No region found from the default AWS config or instance metadata. Defaulting to 'aws-global'
+To avoid seeing this warning, please provide a region in your AWS config or through AWS_REGION environment variable.
```

Comment:
```suggestion
Warning: No region found from the default AWS config or instance metadata. Defaulting to 'aws-global'.
To avoid seeing this warning, please provide a region in your AWS config or through the AWS_REGION environment variable.
```